source/detail/parser/x86_64.cpp

```diff
@@ -128,6 +129,29 @@ namespace smeagle::x86_64 {
     return result.str();
   }
 
+  // Get directionality from argument type
+  std::string getDirectionalityFromType(st::Type *paramType) {
+    auto dataType = paramType->getDataClass();
+    std::string dataTypeStr = st::dataClass2Str(dataType);
+
+    // If it's a pointer, we need to know what it's pointing to!
```

Comment:
The algorithm here is a bit more complicated because we need to handle typedefs and multiple indirections. The code doesn't need to change completely-- just adding more checks.

Reply:
Do you want to talk about it, or just take a shot?